airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/per_partition_cursor.py

```diff
@@ -49,12 +52,15 @@ class PerPartitionCursor(DeclarativeCursor):
     def __init__(self, cursor_factory: CursorFactory, partition_router: PartitionRouter):
         self._cursor_factory = cursor_factory
         self._partition_router = partition_router
-        self._cursor_per_partition: MutableMapping[str, DeclarativeCursor] = {}
+        self._cursor_per_partition: OrderedDict[str, DeclarativeCursor] = OrderedDict()
```

Comment:
can you add a comment in the code explaining why the cursor_per_partition needs to be ordered?